packages/x-components/src/components/result/base-result-image.vue

```diff
@@ -98,8 +153,21 @@
      * @internal
      */
     protected get imageSrc(): string {
-      const image = this.result.images?.find(image => !this.failedImages.includes(image));
-      return image ?? '';
+      return this.loadedImages[
+        !this.showNextImageOnHover || !this.isHovering ? 0 : this.loadedImages.length - 1
+      ];
+    }
+
+    /**
+     * Indicates if the loader should try to load the next image.
+     *
+     * @returns True if it should try to load the next image.
+     *
+     * @internal
+     */
+    protected get shouldLoadNextImage(): boolean {
+      const nextImgIndex = this.showNextImageOnHover && this.userHasHoveredImage ? 2 : 1;
```

Comment:
If talking about indexes these number should be `1` and `0`, ins't it?

Reply:
Nope, the name of the variable is wrong.
They are not indexes, they are the number of images to load.
I will change that name 👍